### PR TITLE
Modify TransformerEncoder masking documentation

### PR DESCRIPTION
**Pull request by sonali-kumari1**
Improved TransformerEncoder masking documentation: Added explicit masking details (using `padding_mask` and custom `attention_mask` arguments) to complement implicit Keras masking (using `mask_zero=True`).
Fixes [#2220 ](https://github.com/keras-team/keras-hub/issues/2220)

### CODE REVIEW

File: keras_hub/src/layers/modeling/transformer_encoder.py [hunk 1]
```diff
@@ -16,9 +16,12 @@ class TransformerEncoder(keras.layers.Layer):
     paper [Attention is All You Need](https://arxiv.org/abs/1706.03762). Users
     can instantiate multiple instances of this class to stack up an encoder.
 
-    This layer will correctly compute an attention mask from an implicit
-    Keras padding mask (for example, by passing `mask_zero=True` to a
-    `keras.layers.Embedding` layer). See the Masking and Padding
+    This layer will compute an attention mask, prioritizing explicitly provided
+    masks(a `padding_mask` or a custom `attention_mask`) over an implicit Keras
```
Comment (mattdangerw) [inline keras_hub/src/layers/modeling/transformer_encoder.py]: space before paren

Comment (sonali-kumari1) [reply keras_hub/src/layers/modeling/transformer_encoder.py]: Hi @mattdangerw, I have added space before parenthesis. Thanks!
